libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFManagedPreferences.m

```diff
@@ -134,4 +139,20 @@ - (BOOL)clearClipboardOnBackground {
     return [self.rawPreferences[kManagedKeyClearClipboardOnBackground] boolValue] || [self shouldDisableExternalPasteDefinedByConnectedApp];
 }
 
+- (void) logAnalyticsEventWithName:name attributes:(NSDictionary *) attributes {
+    if (![SFUserAccountManager sharedInstance].currentUser) {
```

Comment:
Same comment I made on the Android side, I wonder if we could introduce a helper method in the SFSDKSalesforceAnalyticsManager taking name, context and custom attributes.